hook.py

```diff
@@ -19,72 +22,177 @@
 
-    encoded_command = BaseWorld.encode_string(command)
-    encoded_cleanup = BaseWorld.encode_string(cleanup)
+    cleanup, new_payloads = _use_default_inputs(entries, test, platform, test['executor'].get('cleanup_command', ''))
+    payloads.extend(new_payloads)
+    cleanup = _handle_multiline_commands(cleanup)
 
-    return (encoded_command, encoded_cleanup)
+    # TODO handle ART local files in command
+    # eg. https://github.com/redcanaryco/atomic-red-team/blob/a956d4640f9186a7bd36d16a63f6d39433af5f1d/atomics/T1022/T1022.yaml
+    # eg. https://github.com/redcanaryco/atomic-red-team/blob/fa67d8f04188955ec2c3d1532d2212f26ef5e3e8/atomics/T1502/T1502.yaml
+    return (command, cleanup, payloads)
 
 
-async def _save_ability(data_svc, entries, test):
+async def _save_ability(technique_to_tactics, entries, test):
     ability_id = hashlib.md5(json.dumps(test).encode()).hexdigest()
     at_ingested = 0
     at_total = 0
+
+    tactics_li = technique_to_tactics.get(entries['attack_technique'], ['redcanary-unknown'])
+    if len(tactics_li) > 1:
+        tactic = 'multiple'
+    else:
+        tactic = tactics_li[0]
+
+    data = {
+        'id': ability_id,
+        'name': test['name'],
+        'description': test['description'],
+        'tactic': tactic,
+        'technique': {
+            'attack_id': entries['attack_technique'],
+            'name': entries['display_name']
+        },
+        'platforms': {}
+    }
     for p in test['supported_platforms']:
         at_total += 1
         if test['executor']['name'] == 'manual':
             # this test is expected to be run manually by a human, no automation is provided
             continue
-        encoded_command, encoded_cleanup = await _prepare_executor(test)
-        await data_svc.store(
-            Ability(ability_id=ability_id, tactic='redcanary', technique_id=entries['attack_technique'],
-                    technique=entries['display_name'], name=test['name'], description=test['description'],
-                    platform=PLATFORMS.get(p, 'unknown'), executor=EXECUTORS.get(test['executor']['name'], 'unknown'),
-                    test=encoded_command, cleanup=encoded_cleanup, requirements=[], parsers=[], variations=[])
-        )
+
+        command, cleanup, payloads = await _prepare_executor(entries, test, p)
+
+        executor = EXECUTORS.get(test['executor']['name'], 'unknown')
+        platform = PLATFORMS.get(p, 'unknown')
+        ex_dict = {'command': command}
+        ex_dict['payloads'] = payloads
+        ex_dict['cleanup'] = cleanup
+        data['platforms'][platform] = {executor: ex_dict}
+
         at_ingested += 1
 
+    if data['platforms']:  # this might be empty, if so there's nothing useful to save
+        d = DATA_DIR + ('abilities/%s' % tactic)
+        if not os.path.exists(d):
+            os.makedirs(d)
+        file_path = '%s/%s.yml' % (d, ability_id)
+        with open(file_path, 'w') as f:
+            f.write(yaml.dump([data]))
```

Comment:
file_svc provides your disk I/O:
```Python
await self.services.get('file_svc').save_file(filename=<file_name>, payload=<data>, target_dir=<atomic_data_dir>)
```

Reply:
`file_svc` encodes the content because of this line https://github.com/mitre/caldera/blob/6128d748a3253f70f646c0e98f77201bb3d7213e/app/service/file_svc.py#L151, the result is not a valid yaml file.

Reply:
Good catch! That's something we need to patch/address on the core